lib/puppet_blacksmith/forge.rb

```diff
@@ -30,13 +30,34 @@ def push!(name, package = nil)
       unless package
         regex = /^#{username}-#{name}-.*\.tar\.gz$/
         pkg = File.expand_path("pkg")
-        f = Dir.new(pkg).select{|f| f.match(regex)}.last
+        f = Dir.new(pkg).select{|fn| fn.match(regex)}.last
         raise Errno::ENOENT, "File not found in #{pkg} with regex #{regex}" if f.nil?
         package = File.join(pkg, f)
       end
       raise Errno::ENOENT, "File does not exist: #{package}" unless File.exists?(package)
 
-      # login to the puppet forge
+      upload!(package)
+    end
+
+    private
+
+    def upload!(file)
```

Comment:
why is it upload!() instead of upload()?

Reply:
I guess because it has a side effect but I'm not too familiar with ruby conventions.

Reply:
Usually bang! methods are public - they communicate something to the user of an object's API. In this case, it's private, so you could call it dinosaur, and it's fine.

Reply:
Changed to ```upload()```